### PR TITLE
Improving logging and retries of local int test runner

### PR DESCRIPTION
**Pull request by billonahill**
When a local integration test fails to get the test result it needs to also kill the topology. Otherwise retries will fail with "topology already exists" errors.

### CODE REVIEW

File: integration-test/src/python/local_test_runner/main.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 ]
 # Retry variables in case the output is different from the input
 RETRY_COUNT = 5
```
Comment (nlu90) [inline integration-test/src/python/local_test_runner/main.py]: we can increase the RETRY_COUNT to allow the test has more chance while decreasing the RETRY_INTERVAL.

File: integration-test/src/python/local_test_runner/main.py [hunk 2]
```diff
@@ -190,34 +195,35 @@ def submitTopology(heronCliPath, testCluster, testJarPath, topologyClassPath,
       '%s' % (outputFile),
       '%d' % (len(TEST_INPUT))
   ]
-  logging.info("Submitting topology: ")
+  logging.info("Submitting topology: %s", splitcmd)
   logging.info(splitcmd)
```
Comment (nlu90) [inline integration-test/src/python/local_test_runner/main.py]: the splitcmd is logged twice.

File: integration-test/src/python/local_test_runner/main.py [hunk 3]
```diff
@@ -190,34 +195,35 @@ def submitTopology(heronCliPath, testCluster, testJarPath, topologyClassPath,
...
   splitcmd = [
       '%s' % (heronCliPath),
       'kill',
       '--verbose',
       '%s' % (testCluster),
       '%s' % (topologyName),
   ]
-  logging.info("Killing topology:")
+  logging.info("Killing topology: %s", splitcmd)
   logging.info(splitcmd)
```
Comment (nlu90) [inline integration-test/src/python/local_test_runner/main.py]: the splitcmd is logged twice.

File: integration-test/src/python/local_test_runner/main.py [hunk 4]
```diff
@@ -235,11 +240,11 @@ def restartShard(heronCliPath, testCluster, topologyName, shardNum):
       '%s' % (topologyName),
       '%d' % shardNum
   ]
-  logging.info("Killing TMaster command:")
+  logging.info("Killing TMaster: %s", splitcmd)
   logging.info(splitcmd)
```
Comment (nlu90) [inline integration-test/src/python/local_test_runner/main.py]: the splitcmd is logged twice.

**Comment by objmagic** [comment]
FYI, @maosongfu wanted me to improve the local integration test since he spotted some potential flaws. I have to admit that I spent some time on it but felt there is no fundamentally design/programming flaw. Maybe you would like to have a quick look?

File: integration-test/src/python/local_test_runner/main.py (hunk 1 above)
Comment (billonahill) [inline integration-test/src/python/local_test_runner/main.py]: IMO if a test is failing more than 5 times it's either a bad test or our test framework is too brittle. The fact that we have to retry at all is suspect.

File: integration-test/src/python/local_test_runner/main.py (hunk 2 above)
Comment (billonahill) [inline integration-test/src/python/local_test_runner/main.py]: Where do you see the other occurrence? I'm not seeing it. Below we log topologyName.

File: integration-test/src/python/local_test_runner/main.py (hunk 3 above)
Comment (billonahill) [inline integration-test/src/python/local_test_runner/main.py]: where?

Comment (objmagic) [inline integration-test/src/python/local_test_runner/main.py]: `logging.info("Killing topology: %s", splitcmd)`

and again down below

`logging.info(splitcmd)`

**Comment by billonahill** [comment]
Fixed split command dup logging.

@objmagic I see the following areas for improvement:
- There's too much sleeping and retrying. We should be able to determine when we can proceed by watching something or getting an event, but sleep/retry === brittle/slow.
- When a test fails there's little to no info re why it failed. For example look at [this failed CI](https://travis-ci.org/twitter/heron/builds/154576607) where the output file wasn't found. Why did it not appear? What process logs might show us? There's no way to know and if you can't reproduce you're stuck. You can investigate process logs like you would on a running system.
- The tests are too black box. They give input and check output but there's no white-box testing (e.g., assert state of processes or state manager along the way). For scaling int tests we'd need that to verify the state of the system with more granularity along the way.
- The test pattern should be more easily extensible. Currently it's submit/kill process/verify, but you could imagine more variations, like scaling.

**Comment by objmagic** [comment]
@billonahill thx for the suggestions. I'll take care of them at some time.

**Comment by kramasamy** [comment]
@objmagic - can you create issues for @billonahill 's suggestions so that we can track them.

**Pull request merged**